src/main/java/org/mitre/synthea/export/FhirR4.java

```diff
@@ -1455,7 +1456,8 @@ private static BundleEntryComponent allergy(BundleEntryComponent personEntry, Bu
 
     allergyResource.setPatient(new Reference(personEntry.getFullUrl()));
     Code code = allergy.codes.get(0);
-    allergyResource.setCode(mapCodeToCodeableConcept(code, SNOMED_URI));
+    String system = ExportHelper.getSystemURI(code.system);
+    allergyResource.setCode(mapCodeToCodeableConcept(code, system));
```

Comment:
These changes away from using only SNOMED/LOINC/whatever system codes for each resource type may be bigger changes than we think when we go to merge this back into upstream Synthea. Their wiki describes each resource/state type as using a particular system or set of systems; I think this change will remove that constraint: https://github.com/synthetichealth/synthea/wiki/Generic-Module-Framework%3A-Basics#codes